### ISSUE TITLE
Adjust learning rate based on score improvement

### ISSUE THREAD

**Issue by wrapperband** [karpathy/char-rnn#122]
https://github.com/deeplearning4j/deeplearning4j/commit/36ff5acccc3f845e6c29a16cc9b50f2bc3d576e5

This additional feature could be added to auto adjust dropout by rate of change training loss. 
Or an option to stop adjusting at a particular training loss, for instance, retaining flexibility by partial training on some data.
